src/goto-programs/lazy_goto_model.h

```diff
@@ -7,27 +7,44 @@
 #define CPROVER_GOTO_PROGRAMS_LAZY_GOTO_MODEL_H
 
 #include <util/language_file.h>
+#include <util/exit_codes.h>
 
 #include "goto_model.h"
+#include "lazy_goto_functions_map.h"
 #include "goto_convert_functions.h"
 
 class cmdlinet;
+class optionst;
 
 
 /// Model that holds partially loaded map of functions
 class lazy_goto_modelt
 {
+public:
+  typedef std::function<void(
+    const irep_idt &function_name,
+    goto_functionst::goto_functiont &function,
+    symbol_tablet &symbol_table)> post_process_functiont;
+  typedef std::function<bool(goto_modelt &goto_model)> post_process_functionst;
+
 private:
   std::unique_ptr<goto_modelt> goto_model;
   message_handlert &message_handler;
 
 public:
   language_filest language_files;
   symbol_tablet &symbol_table;
-  goto_functionst &goto_functions;
+  const lazy_goto_functions_mapt<goto_programt> goto_functions;
+
+private:
```

Comment:
I'd prefer merging respective private (resp. public) sections.

Reply:
Still applies.

Reply:
This was discussed offline with Peter and a compromise was reached that grouped some parts together but kept the distinct sections, adding some comment to make them clearer.

Reply:
Ok; looking at the final version I'm still a bit confused by "Fields" vs "Public fields" not being grouped together/next to each other. (The commit-by-commit review makes this much harder here as apparently later commits added cleanup.)

Reply:
Peter's concern was that the public API should be in one place. Public fields are, unfortunately, part of that public API and so appear next to public methods after the constructors. Obviously I'd rather not add public fields like this to a new class but I need to maintain compatibility with `goto_modelt` for use in `rebuild_goto_start_function`.

Reply:
If you want to maintain the separation (space-wise) of `private` fields vs `public` fields then at least the comment should also so `Private fields`. I cannot help but say that the fact that comments are required is [suspicious](http://apdevblog.com/comments-in-code/). But also not a blocker, just voicing (too many?) thoughts.